src/main/scala/intellij/haskell/_HaskellLexer.flex

```diff
@@ -229,7 +229,7 @@ nhaddock_start      = {left_brace}{dash}{white_char}?{vertical_bar}
         return HS_ONE_PRAGMA;
     }
 
-    #|\-\} {
+    #|\-\}|\{ {
```

Comment:
@rikvdkleij as you can see I dealt with `{` here so the highlighter doesn't crash when you have pragmas like `{-# { #-}`.
I wonder if there's some ideal way of dealing with this issue.

Also, parsing parenthesis as a child node of the pragma node (instead of parsing them as a sequence) sounds like a better idea.